docs/ja/docs/async.md

```diff
@@ -0,0 +1,393 @@
+
+しかし、WebSocketのサポートを追加するために、非同期Web Python (ASGI) の主な仕様はDjangoで開発されました。
+
+そのような非同期性がNodeJSを人気にした理由です (NodeJSは並列ではありませんが)。そして、プログラミング言語としてのGoの強みでもあります。
+
+そして、それは**FastAPI**で得られるパフォーマンスと同じレベルです。
+
+また、並列処理と非同期処理を同時に実行できるため、テスト済みのほとんどのNodeJSフレームワークよりも高く、Goと同等のパフォーマンスが得られます。Goは、Cに近いコンパイル言語です <a href="https://www.techempower.com/benchmarks/#section=data-r17&hw=ph&test=query&l=zijmkf-1" class="external-link" target="_blank">(Starletteに感謝します)</a>。
+
+### 並行は並列よりも優れていますか？
+
+いや！それはこの話の教訓ではありません。
+
+並行処理は並列処理とは異なります。多くの待機を伴う**特定の**シナリオに適しています。そのため、一般に、Webアプリケーション開発では並列処理よりもはるかに優れています。しかし、すべてではありません。
+
+なので、バランスをとるために、次の物語を想像して下さい:
+
+> 大きくて汚れた家を掃除する必要があります。
+
+*はい、以上です*。
+
+---
+
+待機🕙せず、家の中の複数の場所でたくさんの仕事をするだけです。
+
+あなたはハンバーガーの例のように、最初はリビングルーム、次にキッチンのようにターンを持つことができますが、あなたは何も待機🕙していないので、掃除とクリーニングだけをしていれば、ターンは何にも影響しません。
+
+ターンの有無に関係なく (並行に) 同じ時間がかかり、同じ量の作業が行われることになるでしょう。
+
+しかし、この場合、8人の元レジ係/料理人/現役清掃員👨‍🍳👨‍🍳👨‍🍳👨‍🍳👨‍🍳👨‍🍳👨‍🍳👨‍🍳を手配できて、それぞれ (さらにあなたも) が家の別々の場所を掃除できれば、追加の助けを借りて、すべての作業を**並列**に行い、はるかに早く終了できるでしょう。
+
+このシナリオでは、清掃員 (あなたを含む) のそれぞれがプロセッサーとなり、それぞれの役割を果たします。
+
+また、実行時間のほとんどは (待機ではなく) 実際の作業に費やされ、コンピュータでの作業は<abbr title="Central Processing Unit">CPU</abbr>によって行われます。これらの問題は「CPUバウンド」と言います。
+
+---
+
+CPUバウンド操作の一般的な例は、複雑な数学処理が必要なものです。
+
+例えば:
+
+* **オーディオ** や **画像処理**。
+* **コンピュータビジョン**: 画像は数百万のピクセルで構成され、各ピクセルには3つの値/色があり、通常、これらのピクセルで何かを同時に計算する必要がある処理。
+* **機械学習**: 通常、多くの「行列」と「ベクトル」の乗算が必要です。巨大なスプレッドシートに数字を入れて、それを同時に全部掛け合わせることを考えてみてください。
+* **ディープラーニング**: これは機械学習のサブフィールドであるため、同じことが当てはまります。乗算するのは数値の単一のスプレッドシートではなく、それらの膨大なセットで、多くの場合、それらのモデルを構築および/または使用するために特別なプロセッサを使用します。
+
+### 並行処理 + 並列処理: Web + 機械学習
+
+**FastAPI**を使用すると、Web開発で非常に一般的な並行処理 (NodeJSの主な魅力と同じもの) を利用できます。
+
+ただし、機械学習システムのような **CPUバウンド** ワークロードに対して、並列処理とマルチプロセッシング (複数のプロセスが並列で実行される) の利点を活用することもできます。
+
+さらに、Pythonが**データサイエンス**、機械学習、特にディープラーニングの主要言語であるという単純な事実により、FastAPIはデータサイエンス/機械学習のWeb APIおよびアプリケーション (他の多くのアプリケーションとの) に非常によく適合しています。
+
+本番環境でこの並列処理を実現する方法については、[デプロイ](deployment.md){.internal-link target=_blank}に関するセクションを参照してください。
+
+## `async` と `await`
+
+現代的なバージョンのPythonには、非同期コードを定義する非常に直感的な方法があります。これにより、通常の「シーケンシャル」コードのように見え、適切なタイミングで「待機」します。
+
+結果を返す前に待機する必要があり、これらの新しいPython機能をサポートする操作がある場合は、次のようにコーディングできます。
+
+```Python
+burgers = await get_burgers(2)
+```
+
+カギは `await` です。結果を `burgers`に保存する前に、`get_burgers(2)`の処理🕙の完了を待つ⏸必要があることをPythonに伝えます。これでPythonは、その間に (別のリクエストを受信するなど) 何か他のことができる🔀⏯ことを知ります。
+
+`await` が機能するためには、非同期処理をサポートする関数内にある必要があります。これは、`async def` で関数を宣言するだけでよいです:
+
+```Python hl_lines="1"
+async def get_burgers(number: int):
+    # ハンバーガーを作成するために非同期処理を実行
+    return burgers
+```
+
+...`def` のかわりに:
+
+```Python hl_lines="2"
+# 非同期ではない
+def get_sequential_burgers(number: int):
+    # ハンバーガーを作成するためにシーケンシャルな処理を実行
+    return burgers
+```
+`async def` を使用すると、Pythonにその関数内で `await` 式 (その関数の実行を「一時停止⏸」し、結果が戻るまで他の何かを実行🔀する) を認識しなければならないと伝えることができます。
+
+`async def` 関数を呼び出すときは、`await` しなければなりません。したがって、これは機能しません:
```

Comment:
```suggestion
`async def` 関数を呼び出すときは、「await」しなければなりません。したがって、これは機能しません:
```

Reply:
"`await`" is used as `async def` in original sentence

Reply:
Is it?

> L297:
> When you want to call an `async def` function, you have to "await" it. So, this won't work:

Reply:
ah .. sorry for that, I made mistake.